### PR TITLE
Create an extension to search dbt documentation

### PR DESCRIPTION
**Pull request by b-per**
## Description
This is a new extension to support searching the docs for dbt on https://docs.getdbt.com/
This is similar to the [Vue.js documentation extension](https://www.raycast.com/DavidPascaud/vuejs-documentation) and leverages most of the code from [Search with Algolia](https://www.raycast.com/valentindotxyz/search-with-algolia).

I tried to use Search with Algolia to query the dbt docs but it didn't work as its current implementation doesn't allow nested fields in the configuration.

In relation to capitilazition rules for commands and extensions, dbt should always be lower case.

## Screencast
dbt-documentation-1

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by danielsequeira** [comment]
Hi @b-per 👋 

Thank you so much for your contribution.

I noticed that you are missing a `README.md` file though, could you please add one?

**Comment by b-per** [comment]
Done 😄 

According to the docs I thought the README was only required for extensions that have parameters or require previous setup (which is not the case for this one).

**Comment by danielsequeira** [comment]
Sorry @b-per you are totally right, I missed the fact that was not the case with this extension 🤦

File: extensions/dbt-documentation/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# dbt-documentation Changelog
+
+## [Initial Version] - 2022-07-14
```
Comment (danielsequeira) [inline extensions/dbt-documentation/CHANGELOG.md:3]:
```suggestion
## [Initial Version] - 2022-07-15
```

File: extensions/dbt-documentation/package.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (danielsequeira) [inline extensions/dbt-documentation/package.json:1]: nitpicking here, but the command together with the extension name look a bit verbose. I would suggest maybe call the extension title "dbt" and then the command title "Search Documentation" with the subtitle "dbt".
Raycast 2022-07-15 at 12 40 41

File: extensions/dbt-documentation/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "title": "dbt Documentation Search",
+  "description": "Search the dbt documentation website https://docs.getdbt.com/",
+  "icon": "dbt-icon.png",
+  "author": "b-per",
+  "license": "MIT",
+  "categories": ["Documentation"],
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search dbt Documentation",
```
Comment (danielsequeira) [inline extensions/dbt-documentation/package.json:13]: nitpicking here, but the command together with the extension name look a bit verbose. I would suggest maybe call the extension title "dbt" and then the command title "Search Documentation" with the subtitle "dbt".

File: extensions/dbt-documentation/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { ActionPanel, getPreferenceValues, List, Action, showToast, Toast } from "@raycast/api";
+import { useEffect, useMemo, useState } from "react";
+import algoliasearch from "algoliasearch/lite";
+
+const APPID = "57ZWAOQC7F";
+const APIKEY = "04a5092253f5120fdff2c77b3847d0e1";
```
Comment (danielsequeira) [inline extensions/dbt-documentation/src/index.tsx:6]: why didn't you consider adding this as a preference? I guess this is not an account or user related API key?

File: extensions/dbt-documentation/CHANGELOG.md (hunk 1 above)
Comment (b-per) [reply extensions/dbt-documentation/CHANGELOG.md]: This is updated now

File: extensions/dbt-documentation/package.json (hunk 3 above)
Comment (b-per) [reply extensions/dbt-documentation/package.json]: Completely agree, the suggestion on the command title/subtitle is great and I just changed it.

In regards to the extension name, there is [already one](https://www.raycast.com/zsombor-flds/dbtcloud) related to dbt but focused on the commercial product dbt Cloud, so I thought it'd still make sense to mention that this is about the docs search for dbt.

File: extensions/dbt-documentation/src/index.tsx (hunk 4 above)
Comment (b-per) [reply extensions/dbt-documentation/src/index.tsx:6]: These are the API creds to connect to Algolia for it to return results for the dbt documentation. This is not something that the users need to know/fill-in themselves.

This is the same approach used by the vuejs-documentation extension:
https://github.com/raycast/extensions/blob/0a576d1ce4a411840d8ff1765c0ab3d67d999708/extensions/vuejs-documentation/src/index.tsx#L8-L10

File: extensions/dbt-documentation/package.json (hunk 3 above)
Comment (danielsequeira) [reply extensions/dbt-documentation/package.json]: Ah ok, I didn't check. In that case I guess "dbt Documentation" as the name is fine.

**Review by danielsequeira — APPROVED** [review]

**Comment by danielsequeira** [comment]
Thanks @b-per, before I merge I just noticed that with the name changes your screenshot is outdated. Could you please upload a new one?

**Comment by b-per** [comment]
Sure! Done.

**Review by danielsequeira — APPROVED** [review]

**Pull request merged**